packages/store-sync/src/recs/recsStorage.ts

```diff
@@ -13,25 +12,44 @@ import { isDefined } from "@latticexyz/common/utils";
 import { schemaToDefaults } from "../schemaToDefaults";
 import { defineInternalComponents } from "./defineInternalComponents";
 import { getTableEntity } from "./getTableEntity";
-import { StoreComponentMetadata } from "./common";
+import { ConfigToRecsComponents } from "./common";
 import { tableIdToHex } from "@latticexyz/common";
 import { encodeEntity } from "./encodeEntity";
 import { StorageAdapter } from "../common";
+import { configToRecsComponents } from "./configToRecsComponents";
+import { singletonEntity } from "./singletonEntity";
+import storeConfig from "@latticexyz/store/mud.config";
+import worldConfig from "@latticexyz/world/mud.config";
 
+export type RecsStorageOptions<TConfig extends StoreConfig = StoreConfig> = {
+  world: RecsWorld;
+  config: TConfig;
+};
+
+export type RecsStorageAdapter<TConfig extends StoreConfig = StoreConfig> = {
+  storageAdapter: StorageAdapter<TConfig>;
+  components: ConfigToRecsComponents<TConfig> &
+    ConfigToRecsComponents<typeof storeConfig> &
+    ConfigToRecsComponents<typeof worldConfig> &
+    ReturnType<typeof defineInternalComponents>;
+};
+
+// TODO: make config optional?
 export function recsStorage<TConfig extends StoreConfig = StoreConfig>({
-  components,
-}: {
-  components: ReturnType<typeof defineInternalComponents> &
-    Record<string, RecsComponent<RecsSchema, StoreComponentMetadata>>;
-  config?: TConfig;
-}): StorageAdapter<TConfig> {
-  // TODO: do we need to store block number?
+  world,
+  config,
+}: RecsStorageOptions<TConfig>): RecsStorageAdapter<TConfig> {
+  world.registerEntity({ id: singletonEntity });
 
-  const componentsByTableId = Object.fromEntries(
-    Object.entries(components).map(([id, component]) => [component.id, component])
-  );
+  const components = {
+    ...configToRecsComponents(world, config),
+    ...configToRecsComponents(world, storeConfig),
+    ...configToRecsComponents(world, worldConfig),
+    ...defineInternalComponents(world),
+  };
+  // TODO: do we need to store block number?
```

Comment:
I see this TODO was already there before so not directly related to this PR, but could we add some more context / track this in an issue? This is a more general thing, I feel like for TODO comments to be useful it needs to be clear how to resolve them when eg. when we find them in a couple weeks/months and we don't have the implicit context anymore that we had when writing the TODO

Reply:
agreed on the meta point! been trying to be more diligent about that

I think this specific comment refers to how we track block numbers per row in sqlite/postgres adapters to keep track of when a record was last updated (for the future purposes of streaming changed data to client), but not sure how necessary it is here